### PR TITLE
Implement Hometask 4 - Interfaces

### PR DESCRIPTION
**Pull request by StanchykAF**

### CODE REVIEW

File: src/main/java/com/epam/javacourse/hometask4/Greenhouse.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+import java.util.Arrays;
+
+public class Greenhouse implements Informational, Watered, Changeable {
+
+    private float temperature;
+    private Plant[] plants;
+
+    public Greenhouse(Plant[] plants) {
+        this.plants = Arrays.copyOf(plants, plants.length);
```
Comment (AndreReen) [inline src/main/java/com/epam/javacourse/hometask4/Greenhouse.java]: You don't need to create a copy of an array to set the property.  this.plant = plants will do the trick

File: src/main/java/com/epam/javacourse/hometask4/Greenhouse.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    public Plant[] getPlantsWithType(String type) {
+        Plant[] buf = new Plant[plants.length];
+        int i = 0;
+        for (Plant plant : plants) {
+            if (plant.getType().equalsIgnoreCase(type)) {
+                buf[i] = plant;
+                i++;
+            }
+        }
+        return Arrays.copyOf(buf, i);
```
Comment (AndreReen) [inline src/main/java/com/epam/javacourse/hometask4/Greenhouse.java]: Same issue here and in other methods, just return buf.

**Review by AndreReen — CHANGES_REQUESTED** [review]

File: src/main/java/com/epam/javacourse/hometask4/Greenhouse.java (hunk 2 above)
Comment (StanchykAF) [reply src/main/java/com/epam/javacourse/hometask4/Greenhouse.java]: In this methods (get plants by type, region) I create a buf array of maximum length. But not all elements in this array will have any value. If I just return buf, some values might be null. That's why I return here only the non-null part of buf

Comment (AndreReen) [reply src/main/java/com/epam/javacourse/hometask4/Greenhouse.java]: Creating a copy of an array is an expensive operation in terms of compute power and memory, doing this twice in a row is an overkill.  Having trailing nulls in an arrays doesn't make a difference for a functionality you are implementing here. 
In fact this is the preferred method, as you will understand when you will get to Java Collections and ArrayList implementation.

**Review by AndreReen — APPROVED** [review]

**Pull request merged**